bigbone/src/main/kotlin/social/bigbone/nodeinfo/NodeInfoClient.kt

```diff
@@ -22,13 +22,15 @@ object NodeInfoClient {
     /**
      * Retrieve server information.
      * @param host hostname of the server to retrieve information from
+     * @param scheme The URL scheme
+     * @param port The URL port
      * @return server information, including the name and version of the software running on this server
      * @throws ServerInfoRetrievalException if the request for a server info to [host] was unsuccessful
      */
-    fun retrieveServerInfo(host: String): Server? {
+    fun retrieveServerInfo(host: String, scheme: String, port: Int): Server? {
```

Comment:
We could think about setting default parameter values here to keep the API the same as before for users that are not interested in setting non-default schemes and/or ports:

```suggestion
    @JvmOverloads
    fun retrieveServerInfo(host: String, scheme: String = "https", port: Int = 443): Server? {
```